setup.py

```diff
@@ -18,8 +18,10 @@ def define_extensions(use_openmp):
     # when using the Anaconda distribution (at least on OSX).
     # This could be because Anaconda uses its own assembler?
     # To work around this we do not add -march=native if we
-    # know we're dealing with Anaconda
-    if "anaconda" not in sys.version.lower():
+    # know we're dealing with Anaconda or if CFLAGS is set
+    # with a requestsed arch
+    flags = os.environ.get("CFLAGS", "")
```

Comment:
How about we do the following: if `CFLAGS` is set, ignore all default settings in `setup.py` and rely entirely on `CFLAGS`?

Reply:
If that's want you prefer i don't mind. I'm a little concerned that it would break some existing users in an environment that sets some CFLAGS but they would now miss out on -ffast-math

Reply:
Updated, LMK if this is what you had in mind.

Reply:
That's a fair point. 

What about a new flag, `LIGHTFM_NO_CFLAGS` that simply disables all LightFM cflag settings?

I'm not super knowledgeable about CFLAGS so happy to take your advice here.

Reply:
That sounds reasonable. Updated

Reply:
Any other changes you'd like to see? Or is this good to go?